File: kyonggiyo-infrastructure/src/main/java/kyonggiyo/persistence/image/JpaImageRepositoryImpl.java
package kyonggiyo.persistence.image;

import kyonggiyo.domain.image.Image;
import kyonggiyo.domain.image.ImageType;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
@RequiredArgsConstructor
public class JpaImageRepositoryImpl implements ImageRepository {

    private final ImageJpaRepository imageJpaRepository;

    @Override
    public List<Image> saveAll(List<Image> images) {
        return imageJpaRepository.saveAll(images);
    }

    @Override
    public List<Image> findByImageTypeAndReferenceId(ImageType imageType, Long referenceId) {
        return imageJpaRepository.findByImageTypeAndReferenceId(imageType, referenceId);
    }

    @Override
    public List<Image> findByImageTypeAndReferenceIdIn(ImageType imageType, List<Long> ids) {
        return imageJpaRepository.findByImageTypeAndReferenceIdIn(imageType, ids);
    }

    @Override
    public void deleteById(Long id) {
        imageJpaRepository.deleteById(id);
    }

    @Override
    public void deleteAllByIdInBatch(List<Long> ids) {
        imageJpaRepository.deleteAllByIdInBatch(ids);
    }

}

File: kyonggiyo-infrastructure/src/main/java/kyonggiyo/image/repository/JpaImageRepositoryImpl.kt
package kyonggiyo.image.repository

import kyonggiyo.application.image.domain.entity.Image
import kyonggiyo.application.image.domain.vo.ImageType
import org.springframework.stereotype.Repository

@Repository
class JpaImageRepositoryImpl(
        private val imageJpaRepository: ImageJpaRepository
) : ImageRepository {
    
    override fun saveAll(images: List<Image>): List<Image> {
        return imageJpaRepository.saveAll(images)
    }

    override fun findByImageTypeAndReferenceId(imageType: ImageType, referenceId: Long): List<Image> {
        return imageJpaRepository.findByImageTypeAndReferenceId(imageType, referenceId)
    }

    override fun findByImageTypeAndReferenceIdIn(imageType: ImageType, ids: List<Long>): List<Image> {
        return imageJpaRepository.findByImageTypeAndReferenceIdIn(imageType, ids)
    }

    override fun deleteById(id: Long) {
        imageJpaRepository.deleteById(id)
    }

    override fun deleteAllByIdInBatch(ids: List<Long>) {
        imageJpaRepository.deleteAllByIdInBatch(ids)
    }

}
